### ISSUE TITLE
Smash 64 Layout

### ISSUE THREAD

**Issue by Soundtoxin** [JonnyHaystack/HayBox#39]
There doesn't seem to be a good control layout for Smash 64. X to jump does nothing. If you jump with the C buttons it's harder to hit A to attack since it's all with the thumb. If you use tap jump it makes it difficult to jump and use your neutral special without doing your up special.

**Comment by JonnyHaystack** [comment]
Yeah I never got around to making one and I didn't have a copy of smash 64 when I owned an N64. Definitely would be nice to have though. The way I did the c-pad in the N64Backend is kinda jank as well (it looks at right analog stick X and Y values and interprets those as c-pad buttons) so I might want to extend the output state struct to have some dedicated extra buttons for that kind of thing.

**Comment by garciaErick** [comment]
@Soundtoxin do you have a recommended layout maybe a commit or some pointers to close this would be cool!

**Comment by Soundtoxin** [comment]
Something mirroring the function of the Melee/Ultimate layouts would be good. Jump where it normally is (X) would solve all my old complaints. I don't recall if I had other issues.

**Comment by Soundtoxin** [comment]
https://strategywiki.org/wiki/Super_Smash_Bros./Controls

http://2.bp.blogspot.com/-AcKLl2Hod6E/T1I4Th897QI/AAAAAAAAADg/sKNpSd95kF0/s1600/SSB+Controls.jpg

I found these lists of the original Smash 64 controls. If you look at the layout of the N64 controller, c-left and c-down are where X and Y would be on newer controllers (relative to A and B), so maybe those should be bound to where X and Y are in the Melee layout.

I would probably also try to match newer games' controls with the shield and grab binds, so maybe bind N64 Z to where both L and R would be in Melee Mode so that people used to hitting either button for shielding and rolling can stick to their preferred button, then bind the N64's R button to where Z is in Melee Mode. That way someone can hit the same buttons across every Smash game out of the box and pretty much get what they expect.

It would also be neat if the C buttons could simulate subtle direction + A button for a tilt stick of sorts, but maybe that would have some balance/legality issues. Also some people probably prefer smash stick to tilt stick and might be annoyed if it's hardcoded to the one they don't like.

I would move taunt to the dpad like in newer games since it's not needed often, so in other words bind N64 L button to DPad up, or all the DPad directions. A, B, and Start buttons don't really have anything weird about them so just make them the same spots they'd be in the Melee layout.

I would also like it if it were possible to have some sort of non-Smash layout for playing other N64 games. It would probably work like FGC Mode where you'd change to it after plugging in, and default would still be the Smash one. There are probably some games besides Smash that would be fun to play with these controllers that would benefit from a less Smash-specific layout. I'd maybe match labels, like L is L and R is R, basically applying the Melee Mode binds based on label rather than function. If you can just select Melee Mode after plugging into an N64, maybe no change is needed there. Same for FGC Mode, which can be nice for 2D games that use the DPad.

I haven't got an N64 hooked up again yet so I don't know if the shield/grab stuff was already done in the current layout.
